packages/babylon/test/fixtures/experimental/optional-chaining/optional-super-method/input.js

```diff
@@ -0,0 +1,5 @@
+class a{
+    constructor(){
+        super()?.b;
```

Comment:
This is actually legal. 

```
OptionalExpression[Yield, Await]:
    CallExpression[?Yield, ?Await] OptionalChain[?Yield, ?Await]
    ...elided...

CallExpression[Yield, Await]:
    SuperCall[?Yield, ?Await]
    ...elided...

SuperCall[Yield, Await]:
    super Arguments[?Yield, ?Await]
```

So a `SuperCall` is a `CallExpression`, and `CallExpression` be used as the base of an `OptionalChain`.

Reply:
So, only super properties are not supported, right?

Reply:
Correct.

Reply:
Just had a look at spec, I'm having doubts:- Acc. to spec:- 
```js
OptionalExpression[Yield, Await]:
   MemberExpression[?Yield, ?Await]OptionalChain[?Yield, ?Await]
   CallExpression[?Yield, ?Await]OptionalChain[?Yield, ?Await]
   OptionalExpression[?Yield, ?Await]OptionalChain[?Yield, ?Await]
```
and 
```js
MemberExpression[Yield, Await]:
   PrimaryExpression[?Yield, ?Await]
   MemberExpression[?Yield, ?Await][Expression[+In, ?Yield, ?Await]]
   MemberExpression[?Yield, ?Await].IdentifierName
   MemberExpression[?Yield, ?Await]TemplateLiteral[?Yield, ?Await]
   SuperProperty[?Yield, ?Await]
```

Reply:
So I think we don't need to throw errors in super property/method, am i right?

Reply:
A `SuperProperty` is `super.prop`, which may be the base of an optional chain: `super.prop?.p`. But, the optional chain syntax cannot appear between the `super` the `prop`, only after.

```js
super.prop?.p // legal, base is a `object` is a MemberExpression.

super?.prop // illegal, super is a keyword and may not appear as the `object`

Reply:
This is already implemented. ( though test cases were not provided).. Should i add test cases only in this PR?

Reply:
If that's the case, then just tests to verify.